feat: Fedora 43

RoyalOughtness opened the pull request:
https://github.com/secureblue/secureblue/actions/runs/18612170579

- Fedora 43
- Switch to the DNF module (necessary because `rpm-ostree install` breaks when using custom repos now, due to a bugfix in RPM 6.0. The bug in RPM 4.X was causing the error to not be surfaced.)
- Drop `nvidia-container-toolkit` due to a lack of F43 support
- Drop the `ffmpeg-libs` metapackage in favor of direct package replacement 
- Using a slightly customized `akmodsbuild` to ensure akmods can still be built as container-root
- Stop installing weak deps
- Stop installing ZFS src, devel, debug, and test packages

TODO before merging:
- [ ] add back .prerelease==false for zfs when 2.4 is out 
- [ ] switch `br-next-43` base tags back to `latest`

RoyalOughtness committed:
feat: add f43 branch

RoyalOughtness committed:
fix

RoyalOughtness committed:
fix

RoyalOughtness committed:
fix

RoyalOughtness committed:
use dnf

RoyalOughtness committed:
fixes

RoyalOughtness committed:
fix

RoyalOughtness committed:
fix

RoyalOughtness committed:
dnf

RoyalOughtness committed:
fix

RoyalOughtness committed:
fix

RoyalOughtness committed:
fix

RoyalOughtness committed:
optimizations

RoyalOughtness committed:
add back opencl loader replacement

RoyalOughtness committed:
fixes

RoyalOughtness committed:
fix

RoyalOughtness committed:
testing

RoyalOughtness committed:
fix

RoyalOughtness committed:
fix

RoyalOughtness committed:
fix zfs

RoyalOughtness committed:
fix

RoyalOughtness committed:
fix

RoyalOughtness committed:
fixes

RoyalOughtness committed:
fixsudoremoval

RoyalOughtness committed:
fix zfs

RoyalOughtness committed:
fix

RoyalOughtness committed:
remove unused rpms

RoyalOughtness committed:
fix

RoyalOughtness committed:
add sudo absence validation

RoyalOughtness committed:
remove fuse2-dependent virt-v2v from ujust

HastD commented on files/scripts/installnvidiakmod.sh:
```diff
@@ -28,9 +28,14 @@ fi
+# TODO remove this when fixed upstream
+sed -i '/if \[\[ -w \/var \]\] ; then/,/fi/d' /usr/sbin/akmodsbuild
+dnf install -y --setopt=install_weak_deps=False nvidia-kmod-common nvidia-modprobe
+cp /usr/sbin/akmodsbuild.backup /usr/sbin/akmodsbuild
```
Shouldn't this be `mv` rather than `cp` to avoid leaving the backup in the final image?

RoyalOughtness replied:
Yes, good catch

HastD committed:
fix: preload libno_rlimit_as.so (#1494)

* fix: preload libno_rlimit_as.so

This hooks into `setrlimit` and makes it a no-op when called with
`RLIMIT_AS`, preventing processes from reducing their own address space,
which breaks hardened_malloc.

`/etc/ld.so.preload` is *not* modified in the same way, which means
`setrlimit` calls by PID 1 won't be affected.

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>

* Update desktop-packages.yml

---------

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>
Co-authored-by: RoyalOughtness <129108030+RoyalOughtness@users.noreply.github.com>

RoyalOughtness committed:
fix: flatpak setup should start after graphical session

RoyalOughtness committed:
fix: omit preload conf from initramfs (#1498)

RoyalOughtness committed:
fix: use correct term for file removal (#1500)

RoyalOughtness committed:
fix: check ~/.config for flatpak setup (#1499)

RoyalOughtness committed:
fix: preload exclude file from dracut

HastD committed:
fix(audit): add no_rlimit_as to expected preloads on desktop (#1501)

* chore: add IOT image support to audit script

Add `Image.IOT` variant for `iot-*` images, exempt IOT images from
Trivalent DNS check (same as CoreOS images), fix ruff issues, and make
Image.from_image_ref more precise.

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>

* fix(audit): add no_rlimit_as to expected preloads on desktop

Also add Image.is_server() and Image.is_desktop() methods for
convenience.

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>

RoyalOughtness committed:
fix: downgrade chronyd seccomp filter until upstream fix (#1502)

RoyalOughtness committed:
comment

RoyalOughtness committed:
fix: add no_rlimit_as to the whitelist

HastD commented on files/system/etc/sysconfig/chronyd:
```diff
@@ -1 +1 @@
-OPTIONS=-F1 -r
+OPTIONS=-F2 -r
```
Upstream fix is in testing, so hopefully this can be reverted by the time Fedora 43 is released: https://bodhi.fedoraproject.org/updates/FEDORA-2025-bff3ee13f3

HastD replied:
The fix has been pushed to stable, so we can switch back to `-F1` now.

RoyalOughtness committed:
fix: allow glycin to run sandboxed for thunar (#1514)

RoyalOughtness committed:
feat: add secondary trusted key for future rotation (#1516)

RoyalOughtness committed:
fix: allow trivalent/FM clipboard sharing (#1517)

RoyalOughtness committed:
fix: optimize selinux policy (#1518)

RoyalOughtness committed:
fix: build failure due to unrequired type (#1519)

RoyalOughtness committed:
fix: attribute instead of type (#1520)

RoyalOughtness committed:
revert chronyd sysconfig

RoyalOughtness committed:
fix: no_rlimit_as preloading on server images

RoyalOughtness committed:
fix: hmalloc audit (#1561)

HastD committed:
fix: remove includepkgs line from VPN repos (#1562)

Proton recently split up the package for the VPN, resulting in a new
package not included in the includepkgs list. I'm not sure why such a
list was added in the first place; the VPN repos are disabled unless the
user chooses to enable them, at which point we might as well allow the
user to install any package from the repo.

RoyalOughtness committed:
feat: replace rpm-ostree with patched copr version

RoyalOughtness committed:
Merge branch 'live' into next

RoyalOughtness committed:
fix: rpm-ostree replacement

RoyalOughtness committed:
fix: set sddm to wait for complete udev device initialization (#1567)

RoyalOughtness committed:
chore: various fixes

RoyalOughtness committed:
fix: adjust selinux policy to restore GTK buttons (#1568)

RoyalOughtness committed:
fix: add additional required interfaces

RoyalOughtness committed:
chore: prep for f43 merge

HastD commented on recipes/common/sericea-modules.yml:
```diff
@@ -18,8 +18,10 @@ modules:
     scripts:
       - disablewlrportals.sh
-  - type: rpm-ostree
+      - sddmawaitudevsettle.sh
```
Need to also remove this script from the recipe.

HastD commented on recipes/common/kinoite-modules.yml:
```diff
@@ -39,3 +38,4 @@ modules:
     scripts:
       - setdefaultkdewallpapers.sh
       - systemsettings-standard-malloc.sh
+      - sddmawaitudevsettle.sh
```
Here as well.

HastD commented on recipes/common/cosmic-modules.yml:
```diff
@@ -11,16 +11,18 @@
-        - NetworkManager-openvpn
-        - cosmic-store
-        - cosmic-player
+        packages:
```
Should this have `install-weak-deps: false`?

HastD commented on recipes/common/desktop-packages.yml:
```diff
@@ -10,38 +10,48 @@
-  - wl-clipboard
-  - alsa-firmware
-  - fzf
+  packages:
```
`install-weak-deps: false`?

HastD commented on recipes/common/server-modules.yml:
```diff
@@ -11,24 +11,22 @@
-      - policycoreutils-python-utils
-    remove:
-      - sudo
+      packages:
```
`install-weak-deps: false`?

HastD commented on recipes/common/silverblue-modules.yml:
```diff
@@ -11,31 +11,32 @@
-        - adw-gtk3-theme
-        - gnome-epub-thumbnailer
-        - gnome-tweaks
+        packages:
```
`install-weak-deps: false`?

WavyEbuilder commented on files/scripts/selinux/thunar/thunar.fc:
```diff
@@ -0,0 +1,5 @@
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+/usr/bin/thunar	--	gen_context(system_u:object_r:thunar_exec_t,s0)
```
Missing EOF

RoyalOughtness replied:
why is this an issue?

WavyEbuilder commented on files/scripts/ensuresudoabsent.sh:
```diff
@@ -0,0 +1,23 @@
+then
+    "sudo found. Exiting..."
+    exit 1
+fi
```
Missing EOF

RoyalOughtness replied:
why is this an issue?

WavyEbuilder commented on files/scripts/ensuresudoabsent.sh:
```diff
@@ -0,0 +1,23 @@
+
+set -euo pipefail
+
+if command -v sudo &> /dev/null
```
Some script called sudo or something in PATH would be triggered by this. Why not just check for its existence in `/usr/bin`? Fedora is merged-usr anyway too.

HastD replied:
I think this is fine, if there's a script called `sudo` in the `PATH` during the build process then something has gone wrong anyway so it seems reasonable to error out here.

RoyalOughtness replied:
> Some script called sudo or something in PATH would be triggered by this.

Yes, that's the purpose of this check, to make sure that's not present

RoyalOughtness committed:
comments
